projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

```diff
@@ -696,31 +696,37 @@
 		function processSingleImageGallery() {
 			var images = document.querySelectorAll( 'a img[data-attachment-id]' );
 			Array.prototype.forEach.call( images, function ( image ) {
-				var container = image.parentElement;
+				var link = image.parentElement;
+				var container = link.parentElement;
 
 				// Skip if image was already added to gallery by shortcode.
-				if ( container.parentElement.classList.contains( 'gallery-icon' ) ) {
+				if ( container.classList.contains( 'gallery-icon' ) ) {
 					return;
 				}
 
-				// Skip if the container is not a link.
-				if ( ! container.hasAttribute( 'href' ) ) {
+				// Skip if image is part of a gallery.
+				if ( container.parentElement.classList.contains( 'blocks-gallery-item' ) ) {
```

Comment:
It's not working for me on an existing tiled gallery (the navigation buttons are still inactive) because we need to also include `.tiled-gallery__item`.

Looks like `itemSelector` contains all the possibilities we might need.

I think this should do the trick (maybe):

```js
if ( domUtil.closest( container, itemSelector ) ) {
	return;
}
```

Reply:
Thanks for the catch! I remembered to test the shortcode and then clean forgot about tiled galleries :)

This should be updated now -- `itemSelector` wouldn't quite work because it also catches single images, but this ought to do the trick!